analysis/02_new_model_input/02.2_assign_building_damage_togrid.md

```diff
@@ -0,0 +1,370 @@
+)
+# Not all municipalities are in the damage data set.
+# Not all municipalities in the damage data can be found in the admin 3 shapefile.
+# removing those buildings with incorrect pcode in them
+build_dmg_data_grouped = build_dmg_data[
+    build_dmg_data["pcode"].isin(list(phl_build_municip["ADM3_PCODE"]))
+]
+build_dmg_data_grouped["Totally"].sum()
+```
+
+## Using Area of Municipality
+
+```python
+## Section describing the merging of the
+# north and south buildings from HOTOSM shapefile
+merged_df_points = phl_build_grid.copy()
+merged_df_points = pd.merge(
+    merged_df_points,
+    phl_build_grid.drop("numbuildings", axis=1),
+    on="id",
+    suffixes=(None, "_y"),
+)
+
+merged_df_points["numbuildings"].describe()
+```
+
+```python
+merged_df_points_overlap = merged_df_points.loc[
+    merged_df_points["Centroid"].isin(adm3_perc_ingrid["Centroid"])
+]
+```
+
+```python
+### Section describing the computation of the building damage percentage
+# combining building damage data and grid information
+merged_total_damage_df = adm3_perc_ingrid.merge(
+    build_dmg_data_grouped,
+    left_on="ADM3_PCODE",
+    right_on="pcode",
+    how="left",
+)
+```
+
+```python
+# computing % in each grid
+# totally damaged
+merged_total_damage_df["Totally_Damaged_bygrid"] = (
+    merged_total_damage_df["Municipality Completeness"]
+    * merged_total_damage_df["Totally"]
+)
+```
+
+```python
+merged_total_damage_df[
+    (merged_total_damage_df["pcode"] == "PH025012000")
+    & (merged_total_damage_df["typhoon"] == "Mangkhut")
+]
+```
+
+```python
+test_df = (
+    merged_total_damage_df[["ADM3_PCODE", "Totally_Damaged_bygrid"]]
+    .groupby("ADM3_PCODE")
+    .sum()
+    .reset_index()
+    .merge(
+        build_dmg_data_grouped[["pcode", "Totally"]]
+        .groupby("pcode")
+        .sum()
+        .reset_index(),
+        left_on="ADM3_PCODE",
+        right_on="pcode",
+        how="left",
+    )
+)
+test_df["Diff"] = test_df["Totally_Damaged_bygrid"] - test_df["Totally"]
+test_df["Diff"].describe()
+```
+
+A small difference between the number of totally damaged values.
+This can be explained by the weighting using the area overlap.
+
+```python
+# computing percentage damage
+# merging with building damage data
+merged_perc_damage_df = merged_df_points.merge(
+    merged_total_damage_df, on="id", how="right", suffixes=(None, "_y")
+)
+merged_perc_damage_df[
+    [
+        "id",
+        "Centroid",
+        "ADM3_PCODE",
+        "ADM3_EN",
+        "typhoon",
+        "Year",
+        "Municipality Completeness",
+        "Totally",
+        "numbuildings",
+        "Totally_Damaged_bygrid",
+    ]
+].sort_values(["Totally_Damaged_bygrid"], ascending=False)
+```
+
+This method of assigning weights using area is providing an interesting problem
+due to some grids having a higher number of damaged buildings that actual buildings.
+We changed the approach to use number of buildings in a municipality instead.
```

Comment:
In the final file there are still some cases where there are more buildings damaged than buildings, is that what you're finding here too?